src/app/shared/services/subsigners/magic-subsigner.service.ts

```diff
@@ -94,8 +99,27 @@ export class MagicSubsignerService implements Subsigner {
   }
 
   private forceLogin(opts: SubsignerLoginOpts): Observable<boolean> {
-    return this.getEmail(opts).pipe(
-      switchMap(email => this.subprovider!.auth.loginWithMagicLink({email})),
+    return of(opts).pipe(
+      switchMap(opts => {
+        if (!!opts.socialProvider) {
+          localStorage.setItem('callbackUrl', this.router.constructURL('/callback'))
+          localStorage.setItem('redirectBack', `${getWindow().location.pathname}${getWindow().location.search}`)
+
+          return this.subprovider!.oauth.loginWithRedirect({
+            provider: opts.socialProvider,
+            redirectURI: `${getWindow().location.origin}/callback`,
+          })
+        } else if (!!opts.idToken) {
```

Comment:
What is `idToken`?

Reply:
It's a decentralized ID token. The naming comes from magic (see the [docs](https://magic.link/docs/api-reference/client-side-sdks/web#loginwithcredential)).
They often call it just [idToken](https://magic.link/docs/api-reference/client-side-sdks/web#getidtoken) in their APIs, so I've called it that way. I thought it's OK to name it like this.